bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -236,21 +236,17 @@ public void controlResized(ControlEvent e) {
 	private IPartListener partListener = new IPartListener() {
 		@Override
 		public void partActivated(IWorkbenchPart part) {
-			if (getShell() != null) {
-				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
-			}
+			Display.getDefault().asyncExec(this::adaptToPartActivationChange);
 		}
 
 		@Override
 		public void partDeactivated(IWorkbenchPart part) {
-			// Do nothing
+			Display.getDefault().asyncExec(this::adaptToPartActivationChange);
```

Comment:
better to use `getShell().getDisplay().asyncExec` - here and below

Reply:
https://github.com/eclipse-platform/eclipse.platform.ui/commit/ad63e43557ee5c01ce8857d21d4dc5436905f9bb

NOTE: getShell() is not the "Eclipse Shell" but rather the shell of the Dialog.
Since this asynchronous execution is very quick, it's likely not a problem (?)